Automattic/newspack-block-theme
# feat: mobile menu part proof of concept

### All Submissions:


### Changes proposed in this Pull Request:

This PR adds a mobile menu template part proof of concept by replacing the header.html template part with links to three additional template parts:
* desktop-header.html
* mobile-header.html
* mobile-sidebar.html.

The Desktop Header and Mobile Header are shown/hidden based on screen size, using CSS. The mobile-header.html include a hard-coded button block that opens the mobile-sidebar.html using JavaScript. 

There's a lot that needs to be refined here: 
* Block locking to prevent people from making breaking changes (like deleting the mobile menu toggle, or the desktop header part)
* Better template part naming/organization (so people know not to try to edit the header.html and instead edit the desktop or mobile parts).
* Better icons/buttons for the mobile menu toggles
* ... and probably a bunch of things I'm not thinking about. 

#### Why this approach?

Full site editing is pretty straight forward at the moment: you can add navigation blocks, and set them individually to display as a hamburger always, only on mobile, or never. This is honestly way too simple for the kinds of headers our publishers use. Having separate mobile and desktop header template parts will allow them to be edited individually. And having a mobile sidebar template part will allow publishers to populate it with the same links as the header, or add or remove a few things or (less advisable) replace it with something else entirely.


### How to test the changes in this Pull Request:

1. Apply the PR.  
2. Confirm that the site's header still loads as expected -- it should look something like:

![image](https://user-images.githubusercontent.com/177561/235031880-3764fbff-6bf5-4c6b-b060-da7f27e2cd70.png)

3. Shrink down the browser window.
4. Confirm that you have a mobile menu button, that opens a separate panel -- it should look something like:

![image](https://user-images.githubusercontent.com/177561/235031909-d4af7222-b832-4cb2-b6cb-59dd9a2e48c8.png)

5. Confirm that the 'Close' button in the panel closes the panel, and that clicking the overlay closes the panel.

![image](https://user-images.githubusercontent.com/177561/235031965-0cf41c33-d941-4903-888a-101193aa2195.png)

6. Navigate to the Site Editor by clicking the 'Edit site' link in the admin toolbar.
7. In the right sidebar, click Template Parts; confirm that the list of template parts includes the new Desktop Header, Mobile Header and Mobile Sidebar template parts:

![image](https://user-images.githubusercontent.com/177561/235032037-547be981-0d71-4f84-bb04-86a0e0eaf249.png)

![image](https://user-images.githubusercontent.com/177561/235032114-d806ebb1-0aea-4e40-bf44-ecc049751ebb.png)

8. Confirm that each of these template parts previews correctly, and that you can edit them (try adding blocks, or reordering things). Save each part.
9. Confirm that your edits work on the front end (not totally necessary, but kind of fun). 
10. Navigate back to the Site Editor > Template Parts, and click 'Manage all template parts' at the bottom of the list. 
11. Find each template part you edited in the list (look for the Customized label under the theme name), and use the 'kebab' menu to 'Clear Customizations' -- this makes sure your test site will be loading the versions of these files from the theme in future testing.

![image](https://user-images.githubusercontent.com/177561/235032440-de219e7a-42e7-4ac6-ba23-9a953c3a954c.png)



### Other information:
